src/Featurizers/LagLeadOperatorFeaturizer.h

```diff
@@ -0,0 +1,237 @@
+            return;
+        }
+        Microsoft::Featurizer::RowMajMatrix<std::double_t> ret(_offsets.size(), _horizon);
+        for (size_t row = 0; row < _offsets.size(); ++row) {
+            // col ranges from _horizon - 1 to 0
+            for (std::uint32_t col = _horizon - 1; col < _horizon; --col) {
+                if ((_index - _numPending - col + _offsets[row] < 0)) {
+                    ret(static_cast<std::int64_t>(row), _horizon - 1 - col) = std::nan("");
+                }
+                else {
+                    ret(static_cast<std::int64_t>(row), _horizon - 1 - col) = *std::get<0>(_buffer.range(
+                                                                                        1, 
+                                                                                        static_cast<size_t>(static_cast<std::int64_t>(_buffer.size() - _numPending - col) + _offsets[row])
+                                                                                      ));
+                }
+            }
+        }
+        _buffer.push(std::get<1>(input));
+        ++_index;
+        callback(ret);
+    }
+
+    void flush_impl(typename BaseType::CallbackFunction const & callback) override {
+        if (_numPending != 0) {
+            while(_numPending) {
+                Microsoft::Featurizer::RowMajMatrix<std::double_t> ret(_offsets.size(), _horizon);
+                for (size_t row = 0; row < _offsets.size(); ++row) {
+                    // col ranges from _horizon - 1 to 0
+                    for (std::uint32_t col = _horizon - 1; col < _horizon; --col) {
+                        if (
+                            (_index - _numPending - col + _offsets[row] < 0) ||
+                            (_numPending <= _offsets[row] - col)) {
+                            ret(static_cast<std::int64_t>(row), _horizon - 1 - col) = std::nan("");
+                        }
+                        else {
+                            ret(static_cast<std::int64_t>(row), _horizon - 1 - col) = *std::get<0>(_buffer.range(
+                                                                                                1, 
+                                                                                                static_cast<size_t>(static_cast<std::int64_t>(_buffer.size() - _numPending - col) + _offsets[row])
+                                                                                              ));
+                        }
+                    }
+                }
+                callback(ret);
+                _numPending--;
+            }
+        }
+    }
+};
+
+/////////////////////////////////////////////////////////////////////////
+///  \typedef       LagLeadOperatorEstimator
+///  \brief         Estimator that creates `LagLeadOperatorTransformer`.
+///
+template <typename InputTupleT>
+using LagLeadOperatorEstimator           = Components::InferenceOnlyEstimatorImpl<LagLeadOperatorTransformer<InputTupleT>>;
```

Comment:
You are going to have the same issue as me with this. The InferenceOnlyEstimatorImpl can't pass constructor parameters to the Transformer. I am currently working on fixing the grain transformer. Do you think I should also see if I can apply that fix to this one as well? If not, you will have to do what I did and actually make a full estimator that just does nothing.

Reply:
Per offline discussion, GrainEstimator change will be in a separate PR.